homeworks/08-js-advanced-1/script.js

```diff
@@ -0,0 +1,216 @@
+// 1.Create a function to access the properties of an object.
+
+(() => {
+  const user = {
+    username: "testuser1",
+    preferences: {
+      sound: {
+        maxValue: 50,
+        value: 30,
+      },
+    },
+  };
+  const randomValue = Math.random();
+  const nullValue = null;
+  function pluck(obj, str) {
+    const properties = str.split(".");
+    let currentObj = obj;
+    for (let property of properties) {
+      if (currentObj == null) {
+        return null;
+      }
```

Comment:
"if (currentObj == null)" is a bad one, it checks whether value is falsy, it's better to use strict equal (===). But in this case your code will be broken